realm/realm-library/src/androidTest/java/io/realm/NotificationsTest.java

```diff
@@ -1044,4 +1045,123 @@ public void onChange(AllTypes object) {
             }
         });
     }
+
+    @Test
+    @RunTestInLooperThread
+    public void realmListener_realmResultShouldBeSynced() {
+        final AtomicBoolean changedOnce = new AtomicBoolean(false);
```

Comment:
Since you are actually using two states. It would probably read more easily if you used an AtomicInteger instead and did switches on the number